File: maria/clojure/update/maria_update.clj
; -----------------------------------------------------------------
;
;	maria_update.clj
;
;					Jul/17/2014
;
; -----------------------------------------------------------------
(import '(java.util Date))
(import '(java.text SimpleDateFormat))
(use 'clojure.java.jdbc)
;
(println "*** 開始 ***")

(let [db-host "localhost"
	db-name "city"
	id (first *command-line-args*)
	population (second *command-line-args*)
	today (. (new SimpleDateFormat "yyyy-MM-dd") format (new Date))
	]
(println id)
(println population)
(println today)

  (def db {:classname "org.gjt.mm.mysql.Driver"
           :subprotocol "mysql"
           :subname (str "//" db-host "/" db-name)
           :user "scott"
           :password "tiger"})

  (with-connection db
    (update-values :cities ["id=?" id]
	{:population population :date_mod today})
      ))
(println "*** 終了 ***")
; -----------------------------------------------------------------

File: maria/R/update/maria_update.R
# ---------------------------------------------------------
#	maria_update.R
#
#				Nov/27/2014
#
# ---------------------------------------------------------
print ("*** 開始 ***")
key_in <- commandArgs(trailingOnly=TRUE)[1]
population_in <- commandArgs(trailingOnly=TRUE)[2]

print (key_in)
print (population_in)

library (RMySQL)
mm <- dbDriver ("MySQL")
conn <- dbConnect (mm, dbname="city",user="scott",password="tiger",host="localhost")
#
today = Sys.Date ()
print (today)
command = paste("update cities set POPULATION = ",population_in,",DATE_MOD='",today,"' where ID= '",key_in,"'",sep="")
print (command)
query.result <- dbSendQuery (conn, command)
#
dbDisconnect (conn)
#
print ("*** 終了 ***")
# ---------------------------------------------------------
